settings.php

```diff
@@ -56,5 +56,23 @@
 
 $ADMIN->add('modcmsfolder', $settings);
 
+$settings = new admin_settingpage('mod_cms_search_settings', new lang_string('search:settings', 'mod_cms'));
+if ($ADMIN->fulltree) {
+    $sql = "SELECT mcf.id, mcf.name
+              FROM {customfield_category} mcc
+              JOIN {customfield_field} mcf ON mcf.categoryid = mcc.id
+             WHERE mcc.component = 'mod_cms' AND mcc.area = 'cmsfield' AND mcf.type IN ('text', 'textarea')";
+    $areas = $DB->get_records_sql_menu($sql);
+
+    $settings->add(new admin_setting_configmulticheckbox(
+        'mod_cms/search_area',
```

Comment:
why did you pick to implement search areas like that? How would it work in the search results whey you disable one of the areas here, but documents from this area are already indexed?

Reply:
It can be complicated logic, and I removed this setting.